Automattic/jetpack
# Fix comments on password protected posts

Commenting on password-protected posts on atomic sites is not working.

On atomic sites, comments are posted via iframe.
The iframe points to https://jetpack.wordpress.com/jetpack-comment...
When the comment is submitted there is an inline js that submits the form. This js form submission happens from https://jetpack.wordpress.com/jetpack-comment/ and posts to the comment post endpoint of the atomic site. If the cookie wp-postpass_... is missing from this then we get `comment_on_password_protected` error.

This PR sets same-site none on the cookie to ensure that it is sent out. This way the comments will continue to work on password-protected posts atomic sites.


Fixes https://github.com/Automattic/jetpack/issues/35716


## Proposed changes:

* In case we have `wp-postpass` set the same-site to none

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:






* Sync this PR with atomic site
* Create password protected post
* In a private tab open the password protected post
* Unlock the post with the password
* Check if `wp-postpass_` and `verbum-wp-postpass` are set
* Post a comment
* There should be no errors and the comment should be posted
